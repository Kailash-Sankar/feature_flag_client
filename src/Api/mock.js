function getPackages() {
  return {
    "0": { id: 0, name: "Custom" },
    "1": { id: 1, name: "Silver" },
    "2": { id: 2, name: "Gold" },
    "3": { id: 3, name: "Diamond" }
  };
}

// hardcoded for demo
function getPackageFeatures() {
  return {
    "0": [],
    "1": {
      ad_slates: {
        id: "ad_slates",
        status: 2
      }
    },
    "2": [],
    "3": []
  };
}

export default {
  getPackages,
  getPackageFeatures
};
