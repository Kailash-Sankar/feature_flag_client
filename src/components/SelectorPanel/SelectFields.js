import React from "react";
import { Select } from "antd";
const { Option } = Select;

export function PackageSelect({ pack, packages, onPackageChange }) {
  return (
    <Select
      defaultValue={""}
      style={{ width: 200 }}
      value={pack}
      onChange={onPackageChange}
      placeholder="Select a value"
    >
      {Object.keys(packages).map((k) => (
        <Option key={packages[k].id}>{packages[k].name}</Option>
      ))}
    </Select>
  );
}

export function FeatureSelect({ feature, features, onFeatureChange }) {
  return (
    <Select
      //showSearch
      defaultValue={""}
      style={{ width: 200 }}
      value={feature}
      onChange={onFeatureChange}
      placeholder="Select a value"
    >
      {features.map((f) => (
        <Option key={f.id}>{f.name}</Option>
      ))}
    </Select>
  );
}

export function ProductSelect({ product, products, onProductChange }) {
  return (
    <Select
      style={{ width: 200 }}
      value={product}
      onChange={onProductChange}
      placeholder="Select a value"
    >
      {Object.keys(products).map((k) => (
        <Option key={products[k].id}>{products[k].name}</Option>
      ))}
    </Select>
  );
}

export function CustomerSelect({ customer, customers, onCustomerChange }) {
  return (
    <Select
      //showSearch
      defaultValue={""}
      style={{ width: 200 }}
      value={customer}
      onChange={onCustomerChange}
      placeholder="Select a value"
    >
      {Object.keys(customers).map((k) => (
        <Option key={customers[k].id}>{customers[k].name}</Option>
      ))}
    </Select>
  );
}
