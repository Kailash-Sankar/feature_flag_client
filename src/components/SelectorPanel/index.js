import React from "react";
import * as styles from "./styles.module.less";
import { Spacer } from "@components/Utils";

import {
  PackageSelect,
  FeatureSelect,
  ProductSelect,
  CustomerSelect
} from "./SelectFields";

function SelectorPanel({
  // customer props
  customers,
  customer,
  onCustomerChange,

  // product props
  products,
  product,
  onProductChange,

  //feature props
  features,
  feature,
  onFeatureChange,

  // package props (package is a reserved word)
  packages,
  pack,
  onPackageChange
}) {
  return (
    <div>
      <div className={styles.selectorCo}>
        <span>Customer</span>
        <CustomerSelect
          customer={customer}
          customers={customers}
          onCustomerChange={onCustomerChange}
        />
      </div>

      <Spacer />

      <div className={styles.selectorCo}>
        <span>Product</span>
        <ProductSelect
          product={product}
          products={products}
          onProductChange={onProductChange}
        />
      </div>

      {onFeatureChange && (
        <>
          <Spacer />
          <div className={styles.selectorCo}>
            <span>Feature</span>
            <FeatureSelect
              features={features}
              feature={feature}
              onFeatureChange={onFeatureChange}
            />
          </div>
        </>
      )}

      {onPackageChange && (
        <>
          <Spacer />
          <div className={styles.selectorCo}>
            <span>Package</span>
            <PackageSelect
              packages={packages}
              pack={pack}
              onPackageChange={onPackageChange}
            />
          </div>
        </>
      )}
    </div>
  );
}

export default SelectorPanel;
