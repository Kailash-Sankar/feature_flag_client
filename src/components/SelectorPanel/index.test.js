import { shallow } from "enzyme";
import SelectorPanel from "./index";
import {
  FeatureSelect,
  ProductSelect,
  CustomerSelect,
  PackageSelect
} from "./SelectFields";
import mock from "@mock/data";
import React from "react";

describe("Selector Panel", () => {
  const customerProps = {
    customers: mock.customers,
    customer: mock.customer,
    onCustomerChange: mock.noop
  };
  const productProps = {
    products: mock.products,
    product: mock.product,
    onProductChange: mock.noop
  };
  const featureProps = {
    features: mock.features,
    feature: mock.feature,
    onFeatureChange: mock.noop
  };
  const packageProps = {
    packages: mock.packages,
    pack: mock.pack,
    onPackageChange: mock.noop
  };

  test("customer select", () => {
    const wrapper = shallow(<CustomerSelect {...customerProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  test("product select", () => {
    const wrapper = shallow(<ProductSelect {...productProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  test("feature select", () => {
    const wrapper = shallow(<FeatureSelect {...featureProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  test("package select", () => {
    const wrapper = shallow(<PackageSelect {...packageProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  test("select wrapper", () => {
    const props = {
      ...customerProps,
      ...productProps,
      ...featureProps,
      ...packageProps
    };
    const wrapper = shallow(<SelectorPanel {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
