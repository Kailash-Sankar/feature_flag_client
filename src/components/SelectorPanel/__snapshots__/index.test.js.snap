// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Selector Panel customer select 1`] = `
<Select
  bordered={true}
  choiceTransitionName="zoom"
  onChange={[Function]}
  placeholder="Select a value"
  style={
    Object {
      "width": 200,
    }
  }
  transitionName="slide-up"
  value="101"
>
  <Option
    key="101"
  >
    Alpha
  </Option>
  <Option
    key="102"
  >
    Beta
  </Option>
  <Option
    key="103"
  >
    Gamma
  </Option>
</Select>
`;

exports[`Selector Panel feature select 1`] = `
<Select
  bordered={true}
  choiceTransitionName="zoom"
  defaultValue=""
  onChange={[Function]}
  placeholder="Select a value"
  style={
    Object {
      "width": 200,
    }
  }
  transitionName="slide-up"
  value="ff_one"
>
  <Option
    key="ff_one"
  >
    Feature One
  </Option>
  <Option
    key="ff_two"
  >
    Feature Two
  </Option>
  <Option
    key="ff_three"
  >
    Feature Three
  </Option>
</Select>
`;

exports[`Selector Panel package select 1`] = `
<Select
  bordered={true}
  choiceTransitionName="zoom"
  onChange={[Function]}
  placeholder="Select a value"
  style={
    Object {
      "width": 200,
    }
  }
  transitionName="slide-up"
  value="0"
>
  <Option
    key="0"
  >
    Custom
  </Option>
  <Option
    key="1"
  >
    Silver
  </Option>
  <Option
    key="2"
  >
    Gold
  </Option>
  <Option
    key="3"
  >
    Diamond
  </Option>
</Select>
`;

exports[`Selector Panel product select 1`] = `
<Select
  bordered={true}
  choiceTransitionName="zoom"
  onChange={[Function]}
  placeholder="Select a value"
  style={
    Object {
      "width": 200,
    }
  }
  transitionName="slide-up"
  value="ax"
>
  <Option
    key="ax"
  >
    Ad Insights
  </Option>
  <Option
    key="bx"
  >
    Experience Insights
  </Option>
  <Option
    key="cx"
  >
    Content Insights
  </Option>
  <Option
    key="dx"
  >
    Social Insights
  </Option>
</Select>
`;

exports[`Selector Panel select wrapper 1`] = `
<div>
  <div>
    <span>
      Customer
    </span>
    <CustomerSelect
      customer="101"
      customers={
        Object {
          "101": Object {
            "id": 101,
            "name": "Alpha",
            "products": Array [
              "ax",
              "cx",
            ],
          },
          "102": Object {
            "id": 102,
            "name": "Beta",
            "products": Array [
              "cx",
            ],
          },
          "103": Object {
            "id": 103,
            "name": "Gamma",
            "products": Array [
              "ax",
            ],
          },
        }
      }
      onCustomerChange={[Function]}
    />
  </div>
  <Spacer />
  <div>
    <span>
      Product
    </span>
    <ProductSelect
      onProductChange={[Function]}
      product="ax"
      products={
        Object {
          "ax": Object {
            "id": "ax",
            "name": "Ad Insights",
          },
          "bx": Object {
            "id": "bx",
            "name": "Experience Insights",
          },
          "cx": Object {
            "id": "cx",
            "name": "Content Insights",
          },
          "dx": Object {
            "id": "dx",
            "name": "Social Insights",
          },
        }
      }
    />
  </div>
  <Spacer />
  <div>
    <span>
      Feature
    </span>
    <FeatureSelect
      feature="ff_one"
      features={
        Array [
          Object {
            "attributes": Array [],
            "description": "Enable feature to do something",
            "id": "ff_one",
            "name": "Feature One",
            "product": "ax",
            "status": 2,
          },
          Object {
            "attributes": Array [],
            "description": "Enable feature to do something",
            "id": "ff_two",
            "name": "Feature Two",
            "product": "ax",
            "status": 1,
          },
          Object {
            "attributes": Array [],
            "description": "Enable feature to do something",
            "id": "ff_three",
            "name": "Feature Three",
            "product": "cx",
            "status": 0,
          },
        ]
      }
      onFeatureChange={[Function]}
    />
  </div>
  <Spacer />
  <div>
    <span>
      Package
    </span>
    <PackageSelect
      onPackageChange={[Function]}
      pack="0"
      packages={
        Object {
          "0": Object {
            "id": 0,
            "name": "Custom",
          },
          "1": Object {
            "id": 1,
            "name": "Silver",
          },
          "2": Object {
            "id": 2,
            "name": "Gold",
          },
          "3": Object {
            "id": 3,
            "name": "Diamond",
          },
        }
      }
    />
  </div>
</div>
`;
